paasta_tools/contrib/service_shard_update.py

```diff
@@ -165,11 +166,11 @@ def parse_args():
         dest="autotune_max_disk",
     )
     parser.add_argument(
-        "--iam-role",
-        help="IAM role to use for the shard",
+        "--iam-role-mapping",
+        help="JSON string of deploy_prefix to IAM role mapping to use for the shard",
         required=False,
         type=str,
-        dest="iam_role",
+        dest="iam_role_mapping",
     )
```

Comment:
you'll probably want to have both args exist temporarily - otherwise, there'll be a brief period of time where this is broken since either --iam-role or --iam-role-mapping won't exist in the deployed paasta package